Source/WebKit/WebProcess/Cache/WebCacheStorageConnection.cpp

```diff
@@ -92,18 +85,34 @@ void WebCacheStorageConnection::batchPutOperation(WebCore::DOMCacheIdentifier ca
 
 void WebCacheStorageConnection::reference(WebCore::DOMCacheIdentifier cacheIdentifier)
 {
-    if (!m_connectedIdentifiers.contains(cacheIdentifier))
-        return;
-
+    m_connectedIdentifiers.add(cacheIdentifier);
```

Comment:
Why don't we early return if we already have this identifier in the set?

Reply:
The same DOMCache may be created several times in the same WebProcess (different contexts).
The networking process is supposed to count these instances.
But the new code path is not doing so anymore so we might best do this in WebProcess.
Or maybe we should stop using identifiers and pass ClientOrigin + cache name for every request to better manage network process crashes.